### ISSUE TITLE
Rate limit bugs

### ISSUE THREAD

**Issue by Aziz-AXG** [better-auth/better-auth#392]
Trying to add rate Limit to the auth and found some bugs 

1-  trying to add custom Rules to Phone Number plugin and it don't work but the main rate Limit work fine
```
rateLimit: {
    enabled: true,
    window: 60,
    max: 30,
    customRules: {
      "/phone-number/send-otp": {
        window: 60,
        max: 1,
      },
    },
  },
  ```
  
2- if i add database storage all rate Limit don't work without any error log
```
  rateLimit: {
    enabled: true,
    storage: "database",
    tableName: "rateLimit",
    window: 60,
    max: 30,
    customRules: {
      "/phone-number/send-otp": {
        window: 60,
        max: 1,
      },
    },
  },
```
even after make sure the rate Limit schema is linked 
```
database: drizzleAdapter(db, {
    provider: "sqlite",
    schema: {
      ...schema,
      user: schema.User,
      rateLimit: schema.rateLimit,
    },
  }),
```
and the rateLimit table in db Is empty
Desktop:  win 11,  mode.js 22.6.0

**Comment by jstri** [comment]
Rate limit `customStorage` configuration is either not implemented (at least I couldn't find it) or not being respected.

**Comment by Bekacru** [comment]
hey guys both issues should be fixed on 1.0.10! feel free to re-open if not
